src/BuildScriptGenerator/Node/NodePlatform.cs

```diff
@@ -15,11 +15,14 @@ namespace Microsoft.Oryx.BuildScriptGenerator.Node
 {
     [BuildProperty(
         ZipNodeModulesDirPropertyKey,
-        "Flag to indicate if 'node_modules' folder need to be in zipped form in the output folder. " +
-        "Default is 'false'.")]
+        "Indicates how and if 'node_modules' folder should be compressed into a single file in the output folder. " +
+        "Options are '" + ZipNodeModulesOption + "', and '" + TarGzNodeModulesOption + "'. Default is to not compress. " +
+        "If this option is used, when running the app the node_modules folder must be extracted from this file.")]
     internal class NodePlatform : IProgrammingPlatform
     {
-        internal const string ZipNodeModulesDirPropertyKey = "zip_nodemodules_dir";
+        internal const string ZipNodeModulesDirPropertyKey = "compress_node_modules";
```

Comment:
nit: `ZipNodeModulesDirPropertyKey ` => `CompressNodeModulesPropertyKey `...so that its easier to relate to the property being passed